### PR TITLE
UPDATE: Replace or with inline if

### PR DESCRIPTION
**Pull request by NalinG**
Replacing the previous x or default code with the inline default if x is None else x in 14 files.

Fixes https://github.com/coala-analyzer/coala/issues/1544

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
you heard the @gitmate-bot guy, your lines contain too many characters, we have a limit of 80. please see that worked on for now :)

File: coalib/misc/Exceptions.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 
 def get_exitcode(exception, log_printer=None):
-    log_printer = log_printer or LogPrinter(NullPrinter())
+    log_printer  = LogPrinter(NullPrinter()) if log_printer is None else log_printer
```
Comment (AbdealiJK) [inline coalib/misc/Exceptions.py]: Also extra space before `=`

File: coalib/output/ConfWriter.py [hunk 2]
```diff
@@ -16,7 +16,8 @@ def __init__(self,
                  section_name_surroundings=None,
                  section_override_delimiters=(".",),
                  unsavable_keys=("save",)):
-        section_name_surroundings = section_name_surroundings or {"[": "]"}
+
+ 	section_name_surroundings = {"[": "]"} if section_name_surroundings is None else section_name_surroundings
```
Comment (AbdealiJK) [inline coalib/output/ConfWriter.py]: It looks like you've removed a space from the front of the line (which messes up indentation everywhere)

File: coalib/parsing/ConfParser.py [hunk 3]
```diff
@@ -15,7 +15,8 @@ def __init__(self,
                  key_delimiters=(',', ' '),
                  section_name_surroundings=None,
                  remove_empty_iter_elements=True):
-        section_name_surroundings = section_name_surroundings or {"[": "]"}
+        
+	section_name_surroundings = {"[": "]"} if section_name_surroundings is None else section_name_surroundings
```
Comment (AbdealiJK) [inline coalib/parsing/ConfParser.py]: Space removed before like

Comment (AbdealiJK) [inline coalib/parsing/ConfParser.py]: The space removed before line has happened in a lot of places  - please check them all out (not commenting everywhere)

File: tests/settings/FunctionMetadataTest.py [hunk 4]
```diff
@@ -98,8 +98,9 @@ def check_function_metadata_data_set(self,
                                          retval_desc="",
                                          non_optional_params=None,
                                          optional_params=None):
-        non_optional_params = non_optional_params or {}
-        optional_params = optional_params or {}
+
```
Comment (AbdealiJK) [inline tests/settings/FunctionMetadataTest.py]: @Makman2 do we really want this change in cases like this ?
if it's falsey it's enough for us to use the alternative right ?

File: coalib/coala_main.py [hunk 5]
```diff
@@ -82,7 +83,8 @@ def run_coala(log_printer=None,
             yielded, yielded_unfixed, results[section_name] = (
                 simplify_section_result(section_result))
 
-            yielded_results = yielded_results or yielded
+            
+	    yielded_results = yielded if yielded_results is None else yielded_results
```
Comment (Adrianzatreanu) [inline coalib/coala_main.py]: you also modified the indentation here of what im aware

**Comment by Adrianzatreanu** [comment]
as an overview, to fix this you would need to:
1. fix the 80 characters problem and make them on two lines instead of one
2. fix the indentation, our coding style is PEP8, you can read from there if you want, but it's mostly different spacings here, please respect the ones we had :)
3. don't add extra newlines where it isn't needed, just check where we didnt have and keep it that way :)

**Comment by Adrianzatreanu** [comment]
this was just an overview for you to see the review easier, if you need any kind of help, please feel free to message us on [gitter](https://gitter.im/coala-analyzer/coala) :)

**Comment by NalinG** [comment]
Okay, really messed it up.
Thanks @Adrianzatreanu , will fix it up by tomorrow

**Comment by NalinG** [comment]
Hi, I am at a loss.
I removed the extra newlines, verified the line length and checked for trailing white spaces according to PEP8, but it still is not passing all the tests.
Can someone tell me where am I wrong...

**Comment by Makman2** [comment]
@NalinG uh you had some merge conflicts and did not resolve them. You can see that inside the circle-ci build log.
Ah just seeing it's your last merge commit. Please rebase interactively, and because of the introduced conflict-markers (the `>>>>>>> HEAD` stuff) you should maybe explicitly exclude the merge commit inside the rebase script^^

**Comment by Makman2** [comment]
@NalinG you need more help? :) feel free to contact us here or at gitter :+1:

**Comment by NalinG** [comment]
Sorry but due to my exams I wont be able to contribute anything.
You can assign it to someone else if anyone is interested.

**Comment by sils1297** [comment]
closing due to low importance, commits are still visible from the issue. THanks @NalinG ! :)

**Pull request closed without merging**